python/packages/autogen-ext/src/autogen_ext/storage/_base.py

```diff
@@ -0,0 +1,369 @@
+from typing import (
+    Any,
+    Callable,
+    List,
+    Mapping,
+    Optional,
+    Protocol,
+    Sequence,
+    Tuple,
+    Union,
+    runtime_checkable,
+)
+
+from pydantic import BaseModel
+
+Metadata = Union[Mapping[str, Any], None]
+Vector = Union[Sequence[float], Sequence[int]]
+ItemID = Union[str, int]
+
+
+class Document(BaseModel):
+    """Define Document according to autogen 0.4 specifications."""
+
+    id: ItemID
+    content: Optional[str] = None
```

Comment:
why assuming string?

Reply:
This is the same representation used in 0.2 and I believe it assumes the content has already been preprocessed to a string format.  More complex data types can lead to more complex operations in general. Would you suggest something different?

Reply:
the vector store can have a payload object, which is usually a json. Unless you know the schema mapping returning string is a problem, maybe we can use dictionary or some other model?

Reply:
I don't think this is meant to capture the full payload but only the main text content of the associated document and the transformation between autogen doc and vectordb doc is handled in each implementation. The pydantic model is defined with `arbitrary_types_allowed` which should allow implementations to unpack additional attributes as needed at the document level instead of the content. Would that work for the  scenario you have in mind?